app/validators/routing/new_when_rule_validator.py

```diff
@@ -101,6 +105,9 @@ def validate_rule(self, rule):
 
         if operator_name == OPERATOR_DATE:
             return TYPE_DATE
+
+        if operator_name == OPERATOR_COUNT:
+            self._validate_count_operator(rule)
```

Comment:
This needs to be done the same way as the date one. You need to first validate the operator then return the `type` for the operator. At the moment, this code would raise errors against your runner schema. This also highlight missing tests, we have tests to validate types for dates, so we need to do the same for count.